### PR TITLE
internal/v4: implement meta/supported-series

### PR DESCRIPTION
**Pull request by rogpeppe**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1185/
Test FAILed.

File: docs/API.md [hunk 1]
```diff
@@ -1065,6 +1066,26 @@ Response body:
...
+This path returns the set of series supported by the given
+charm. This endpoint is appropriate for charms only.
+
+```go
+type SupportedSeriesResponse struct {
+    SupportedSeries []string
+}
+```
+
+Example: `GET wordpress/meta/hash256`
```
Comment (fabricematrat) [inline docs/API.md]: why hash256 here ?

**Comment by fabricematrat** [comment]
:+1:  with one question on the doc

File: docs/API.md (hunk 1 above)
Comment (frankban) [inline docs/API.md]: This example and the response body were not changed after the copy/paste.

**Comment by frankban** [comment]
:+1: with a doc trivial.
Thank you!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1186/
Test PASSed.

File: docs/API.md (hunk 1 above)
Comment (rogpeppe) [inline docs/API.md]: funnily enough, the changes were sitting unsaved in my editor :)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1188/
Test PASSed.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1189/
Test PASSed.

**Pull request merged**